packages/core/src/app/serialization/examples/simple-state-channel.ts

```diff
@@ -0,0 +1,81 @@
+import { BigNumber, Property, WitnessFactory } from '../../../types'
+import {
+  AndDecider,
+  ForAllSuchThatDecider,
+  MessageEqualsDecider,
+  MessageEqualsInput,
+  MessageInvalidatedDecider,
+  MessageInvalidatedInput,
+  MessageInvalidatedWitness,
+} from '../../ovm/deciders'
+import {
+  SignedByDecider,
+  SignedByInput,
+} from '../../ovm/deciders/signed-by-decider'
+import { OrDecider } from '../../ovm/deciders/or-decider'
+import { SignedByQuantifier } from '../../ovm/quantifiers/signed-by-quantifier'
+import { ThereExistsSuchThatDecider } from '../../ovm/deciders/there-exists-such-that-decider'
+
+export interface AddressBalance {
+  [address: string]: BigNumber
+}
+
+export interface MessageInvalidatedClaim extends Property {
+  decider: MessageInvalidatedDecider
+  input: MessageInvalidatedInput
+}
+
+export type MessageInvalidatedWitnessFactory = (
+  input: any
+) => MessageInvalidatedWitness
+
+export interface MessageInvalidatedOrCurrentClaim extends Property {
+  decider: OrDecider
+  input: {
+    properties: [
+      {
+        decider: ThereExistsSuchThatDecider
+        input: {
+          quantifier: SignedByQuantifier
+          quantifierParameters: any
+          propertyFactory: MessageInvalidatedClaim
+          witnessFactory: MessageInvalidatedWitnessFactory
```

Comment:
A little confused by there being a witness here in the input.  Generally witnesses are used to make a decision for some property, but the witness is not a part of the input to that property.

Reply:
That definitely makes sense. The witness factory is just a mechanism to pass witnesses to nested deciders. It's not a witness per se, but a function that will generate a witness in the context in which it's used, but your point is still valid.

Either way, I'm interested to hear your thoughts if you think we can make this better!